docs: cover 5.4.0 release changelog (#102)
neo4j/docs-kafka-connector: modules/ROOT/pages/sink/cud.adoc

MATCH (start)-[r:RELATED_TO {id: $ids.id}]->(end)
DELETE r
----

== Batching of CUD events

Starting from version 5.4.0, a batched sink handler is available for the CUD strategy to provide better throughput when processing CUD events.
This handler processes messages in batches and applies them to the database more efficiently.

There are currently two batching strategies used by the CUD sink handler:

> . APOC Core based strategy: This strategy uses the `apoc.cypher.doIt` procedure to execute batches of Cypher statements generated from the CUD events.

Reviewer: ```suggestion
. APOC Core strategy: This strategy uses the `apoc.cypher.doIt` procedure to execute batches of Cypher statements generated from the CUD events.
```
Should be clear enough

. Native strategy: This strategy generates a single Cypher statement containing multiple subqueries for each message in the batch, and executes it using the standard Cypher execution engine.
The maximum number of subqueries included in a single batch when using the native strategy can be configured using the `neo4j.max-batched-queries` setting, which defaults to `50`.

> Connector prefers APOC Core `apoc.cypher.doIt` based batching strategy when available, falling back to the native strategy when APOC Core is not available on the target database.

Reviewer: ```suggestion
The connector defaults to the APOC core batching strategy when available, falling back to the native strategy when APOC Core is not available on the target database.
```

Batch size, which applies to both batching strategies, can be configured using the `neo4j.batch-size` setting, which defaults to `1000`.

== Exactly-once semantics

Starting from version 5.4.0, the CUD sink handler can be configured to achieve exactly-once processing guarantees by keeping track of the last successfully processed message offset in the database.
This is achieved by storing a dedicated node representing the offset of the last successfully processed message.

> Use `neo4j.eos-offset-label` setting to specify a label name to attach to the offset node.

Reviewer: ```suggestion

Use the `neo4j.eos-offset-label` setting to specify a label name to attach to the offset node.
```
Note that this adds a newline to create a new paragraph.

This setting is empty by default, meaning that exactly-once semantics are disabled and the connector will not keep track of processed message offsets in the target database, meaning that at-least-once processing guarantees are provided.

[IMPORTANT]
====

> Make sure that the configured label has a NODE KEY constraint defined on the 'strategy', 'topic' and 'partition' properties.

Reviewer: ```suggestion
Make sure that the configured label has a node key constraint defined on the `strategy`, `topic` and `partition` node properties.
```

> Assuming your `neo4j.eos-offset-label` is set to `__KafkaOffset`, the following constraint should be created on the target database:

Reviewer: ```suggestion
For example, if your `neo4j.eos-offset-label` is set to `__KafkaOffset`, the following constraint must appear on the target database:
```
